lib/checken/schema.rb

```diff
@@ -32,48 +32,11 @@ def configure(&block)
     # @param permission_path [String]
     # @param user [User]
     # @param object [Object]
-    def check_permission!(permission_path, user_proxy, object = nil)
```

Comment:
This diff looks big but I just extracted the logic as-is down into the new `handle_strict_permission_check!` method below.